.github/workflows/sql.yml

```
@@ -0,0 +1,71 @@
+name: SQL
+
+on:
+  push:
+    branches:
+      - main
+  pull_request: {}
+
+jobs:
+  mysql:
+    name: ${{ matrix.database.name }}
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        database:
+          - {name: MySQL 5.7,      image: "mysql:5.7"}
+          - {name: MySQL 8.0,      image: "mysql:8.0"}
+          - {name: MySQL latest,   image: "mysql:latest"}
+          - {name: MariaDB 10.1,   image: "mariadb:10.1"}
+          - {name: MariaDB 10.2,   image: "mariadb:10.2"}
+          - {name: MariaDB 10.3,   image: "mariadb:10.3"}
+          - {name: MariaDB 10.4,   image: "mariadb:10.4"}
+          - {name: MariaDB 10.5,   image: "mariadb:10.5"}
+          - {name: MariaDB 10.6,   image: "mariadb:10.6"}
+          - {name: MariaDB 10.7,   image: "mariadb:10.7"}
+          - {name: MariaDB 10.11,  image: "mariadb:10.11"}
+          - {name: MariaDB 11.0,   image: "mariadb:11.0", compat: 'false'}
+          - {name: MariaDB latest, image: "mariadb:latest", compat: 'false'}
+
+    env:
+      ICINGAGOLIBRARY_TESTS_DB_TYPE: mysql
+      ICINGAGOLIBRARY_TESTS_DB: icinga_unittest
+      ICINGAGOLIBRARY_TESTS_DB_USER: root
+      ICINGAGOLIBRARY_TESTS_DB_PASSWORD: password
+      ICINGAGOLIBRARY_TESTS_DB_HOST: 127.0.0.1
+      ICINGAGOLIBRARY_TESTS_DB_PORT: 3306
+
+    services:
+      mysql:
+        image: ${{ matrix.database.image }}
+        env:
+          MYSQL_ROOT_PASSWORD: password
+          MYSQL_DATABASE: icinga_unittest
+          MARIADB_ROOT_PASSWORD: password # mariadb >= 11
+          MARIADB_DATABASE: icinga_unittest # mariadb >= 11
+        # Wait for the containers to become ready
+        options: >-
+          --health-cmd ${{ matrix.database.compat != 'false' && '"mysqladmin ping"' || '"healthcheck.sh --connect --innodb_initialized"' }}
```

Would something like the following also work, i.e. provide a default `--health-cmd` and allow to override it the matrix?

```suggestion
          --health-cmd "${{ matrix.database.healthcmd || 'mysqladmin ping' }}"
```

---

Maybe I didn't test it, but why? That way you would have to define `healthcmd` for every MariaDB version that is not compatible with MySQL.

---

It's just that `compat: 'false'` in the matrix definition is pretty meaningless. My suggestion would be to try to make it more expressive.

---

> It's just that compat: 'false' in the matrix definition is pretty meaningless.

How is that "meaningless"? It just indicates whether that specific mariadb image is fully mysql-compatible or not, and I don't consider that as "meaningless" at all. If you like, I can add a few sentences as a comment but in any case, I find having `compat: 'false'` in the matrix far better than defining the entire health command every time.